setup.py

```diff
@@ -1,8 +1,53 @@
-from distutils.core import setup
-from Cython.Build import cythonize
-import numpy
-
-setup(
-    ext_modules=cythonize("*.pyx", language="c++"),
-    include_dirs=[numpy.get_include()]
-)
+from setuptools import Extension, find_packages, setup
+
+if __name__ == '__main__':
+  # see https://stackoverflow.com/questions/2379898/make-distutils-look-for-numpy-header-files-in-the-correct-place
+  # for the approach to pushing numpy an cython dependencies into extension building only
```

Comment:
72 characters long, please.

Maybe I should add a lint checker to this project sometime...

Reply:
Sounds good!